java/client/src/org/openqa/selenium/support/pagefactory/ByChained.java

```diff
@@ -48,10 +48,21 @@ public ByChained(By... bys) {
 
   @Override
   public WebElement findElement(SearchContext context) {
-    List<WebElement> elements = findElements(context);
-    if (elements.isEmpty())
-      throw new NoSuchElementException("Cannot locate an element using " + toString());
-    return elements.get(0);
+    if (bys.length == 0) {
+      throw new NoSuchElementException("No Bys were specified in this " + getClass().getSimpleName());
+    }
+
+    WebElement elem = null;
+    for (By by : bys) {
+      boolean isFirstIteration = elem == null;
+      if (isFirstIteration) {
+        elem = by.findElement(context);
+      } else {
+        elem = elem.findElement(by);
+      }
+    }
+    assert elem != null; // NoSuchElement should have already been thrown
```

Comment:
Ah I see now! Thanks for the further explanation :)
So I've always been misunderstanding the behaviour of ByChained, damn.
The thing is that by reading the javadoc

> This class will find all DOM elements that matches each of the locators in sequence,

I understood that it behaved like a logical AND of Bys, and as such that it would "fail" if any one of the By failed.
Should it have been "each and every" if it had the meaning I got? Not sure as I'm not english mothertongue.

I'm wondering if it's only me to have misunderstood or there are (many?) others.
If it's only me, then fine, I can just add a new class with the 'AND' behavior. Also I would like to add a line to the javadoc, explaining the thing, as well as maybe an example, why not.

What do you think?